ext/oci8/oci8.stub.php

```diff
@@ -89,6 +89,10 @@ function oci_lob_is_equal(OCILob $lob1, OCILob $lob2): bool {}
 
 function oci_lob_export(OCILob $lob, string $filename, ?int $offset = null, ?int $length = null): bool {}
 
+function oci_lob_write_temporary(OCILob $lob, string $data, int $type = OCI_TEMP_CLOB): bool {}
```

Comment:
Apparently these functions haven't been exposed before. Not sure I should now do it, or rather mark the OO coiunterparts with `@implementation-alias` (I'm happy that we are finally able to profit from this :) ).

Reply:
We should check this with @cjbj. Clearly these were supposed to exist originally, but I don't know it it makes sense to add them now or not.

Reply:
OCI8 only documents two LOB functions at https://www.php.net/manual/en/ref.oci8.php These are the LOB functions that take two LOB arguments. Most access is (now) via [OO](https://www.php.net/manual/en/class.OCI-Lob.php). A few more LOB functions are documented in the obsolete alias section https://www.php.net/manual/en/oldaliases.oci8.php

Does this help? I'm not sure what your tooling is doing. I'd be happy to set up a call, if discussion would be easiest.

Reply:
Huh, that's even more peculiar than I thought... Let me describe the situation as it is right now and has been for a long time -- I checked that it's the same on the PHP-7.0 branch at least, possibly also before:

 * We have the OCI-Lob OO API with documented methods.
 * All of these methods have a corresponding `oci_lob_*` function, with the exception of `oci_write_temporary` and `oci_lob_close` (what we found out here).
 * The documentation does not list any of these `oci_lob_*` functions! Only oci_lob_copy and oci_lob_is_equal, which are separate functions, not part of the OO API.
 * The documentation does list `ociwritetemporarylob` and `ocicloselob` as obsolete aliases -- but those aliases don't actually exist...

Having this in mind, especially the fact that the current `oci_lob_*` procedural API is undocumented, I think the right action here would be:

 * For PHP 8.0 and the purposes of this PR: Do not add `oci_write_temporary` and `oci_lob_close`. Change the implementation instead so these are directly implemented only as methods.
 * For the future: Mark the `oci_lob_*` API as deprecated (with the exception of oci_lob_copy and oci_lob_is_equal again). Given that these are not even documented, it seems sensible to remove them in the future.

Does that make sense?

Reply:
OCI8 has a long history.

What does changing the oci_write_temporary and oci_lob_close buy us now?  I'd hate to break anything someone happened for some reason to be using.

I'm happy to mark the oci_lob* API as deprecated right now in 8.0 (except for _copy & _is_equal) but am not immediately sure how to do that and keep the object API not-deprecated and how the stubs stuff fits in.

Reply:
> What does changing the oci_write_temporary and oci_lob_close implementation to methods buy us now

This won't have any effect to users, it's only an implementation detail. What I did is only removed the alias (https://github.com/php/php-src/pull/6335/commits/f695ad3d7fe1311b97b1279ec7403925a12a9fca#diff-2d27846334079384fd40d9d1adc5a2e546b36cee995e1444d156b6a4b20decf3L662), and changed `PHP_FUNCTION` to `PHP_METHOD`. The only effect of this change is that we 1) won't report a non-existent alias function in the stubs 2) make the implementation a bit more straightforward by eliminating the unneeded alias

> I'm happy to mark the oci_lob* API as deprecated right now in 8.0

I think we don't have to hurry with it, especially because the release of php 8.0 is right in the corner. I mean, as PHP 8.0 is already branched off, you could commit the deprecation to the master branch, even right now.

> I'm not sure what your tooling is doing.

The change I implemented in this PR is a verification that alias functions/methods and their aliased counterparts have the same parameters (name, type, default value). And this change brought up the glitch with the OCILob alias. :)